onnxruntime/test/providers/cpu/math/gemm_test.cc

```diff
@@ -114,162 +124,201 @@ static void TestGemmTrans(bool b_is_initializer) {
+                   {1.0, 2.0, 3.0, 4.0,
+                    -1.0, -2.0, -3.0, -4.0});
+  test.AddInput<T>("B", {3, 4}, std::vector<T>(12, 1.0));
+  test.AddInput<T>("C", {2, 1}, std::vector<T>(2, 1.0));
+  test.AddOutput<T>("Y", {2, 3},
+                    {11.0, 11.0, 11.0,
+                     -9.0, -9.0, -9.0});
 #if defined(OPENVINO_CONFIG_GPU_FP16) || defined(OPENVINO_CONFIG_GPU_FP32)
   test.Run(OpTester::ExpectResult::kExpectSuccess, "", {kOpenVINOExecutionProvider});  // OpenVINO: Temporarily disabled due to accuracy issues
 #else
   test.Run();
 #endif
 }
 
-TEST(GemmOpTest, GemmAlphaBeta) {
+TEST(GemmOpTest, GemmTransB_1) {
+  TestGemmTransB_1<float>();
+  TestGemmTransB_1<double>();
+}
+
+template <typename T>
+void TestGemmAlphaBeta() {
   OpTester test("Gemm");
 
   test.AddAttribute("transA", (int64_t)0);
   test.AddAttribute("transB", (int64_t)0);
   test.AddAttribute("alpha", 0.5f);
   test.AddAttribute("beta", 2.0f);
 
-  test.AddInput<float>("A", {2, 4},
-                       {1.0f, 2.0f, 3.0f, 4.0f,
-                        -1.0f, -2.0f, -3.0f, -4.0f});
-  test.AddInput<float>("B", {4, 3}, std::vector<float>(12, 1.0f));
-  test.AddInput<float>("C", {3}, std::vector<float>(3, 1.0f));
-  test.AddOutput<float>("Y", {2, 3},
-                        {7.0f, 7.0f, 7.0f,
-                         -3.0f, -3.0f, -3.0f});
+  test.AddInput<T>("A", {2, 4},
+                   {1.0, 2.0, 3.0, 4.0,
+                    -1.0, -2.0, -3.0, -4.0});
+  test.AddInput<T>("B", {4, 3}, std::vector<T>(12, 1.0));
+  test.AddInput<T>("C", {3}, std::vector<T>(3, 1.0));
+  test.AddOutput<T>("Y", {2, 3},
+                    {7.0, 7.0, 7.0,
+                     -3.0, -3.0, -3.0});
 #if defined(OPENVINO_CONFIG_GPU_FP16) || defined(OPENVINO_CONFIG_GPU_FP32)
   test.Run(OpTester::ExpectResult::kExpectSuccess, "", {kTensorrtExecutionProvider, kOpenVINOExecutionProvider});  // OpenVINO: Temporarily disabled due to accuracy issues
 #else
   test.Run(OpTester::ExpectResult::kExpectSuccess, "", {kTensorrtExecutionProvider});  //TensorRT: Seg fault in parser
 #endif
 }
 
-TEST(GemmOpTest, GemmNaN) {
+TEST(GemmOpTest, GemmAlphaBeta) {
+  TestGemmAlphaBeta<float>();
+  TestGemmAlphaBeta<double>();
+}
+
+template <typename T>
+void TestGemmNaN() {
   OpTester test("Gemm");
 
   test.AddAttribute("transA", (int64_t)0);
   test.AddAttribute("transB", (int64_t)0);
   test.AddAttribute("alpha", 1.0f);
   test.AddAttribute("beta", 0.0f);
 
-  test.AddInput<float>("A", {2, 4},
-                       {1.0f, 2.0f, 3.0f, 4.0f,
-                        -1.0f, -2.0f, -3.0f, -4.0f});
-  test.AddInput<float>("B", {4, 3}, std::vector<float>(12, 1.0f));
-  test.AddInput<float>("C", {2, 3}, std::vector<float>(6, 1.0f));
-  test.AddOutput<float>("Y", {2, 3},
-                        {10.0f, 10.0f, 10.0f,
-                         -10.0f, -10.0f, -10.0f});
+  test.AddInput<T>("A", {2, 4},
+                   {1.0, 2.0, 3.0, 4.0,
+                    -1.0, -2.0, -3.0, -4.0});
```

Comment:
I think you need to keep the 'f' ,otherwise by default it is double. Converting from double to float is not lossless, so it may generate warnings?

Reply:
All examples use integer written as double. The conversion into float is the same for small integers. Adding a f in this case should not have any impact. I can put them back.

Reply:
See: https://docs.microsoft.com/en-us/cpp/cpp/numeric-boolean-and-pointer-literals-cpp?view=msvc-160

"Floating-point literals default to type double", unless you add a suffix 'f'. 

The conversion from float to double is always safe, but not vice versa. So, if float is enough, it's better to define them as float.